elements/pf-select/pf-select.ts

```diff
@@ -0,0 +1,517 @@
+   */
+  @property({ reflect: true, attribute: 'match-anywhere', type: Boolean }) matchAnywhere = false;
+
+  /**
+   * whether multiple items can be selected
+   */
+  @property({ reflect: true, attribute: 'multi-selectable', type: Boolean }) multiSelectable = false;
+
+  /**
+   * whether listbox is always expanded
+   */
+  @property({ reflect: true, attribute: 'expanded', type: Boolean }) expanded = false;
+
+  /**
+   * Indicates initial popover position.
+   * There are 6 options: `bottom`, `top`, `top-start`, `top-end`, `bottom-start`, `bottom-end`.
+   * Default is `bottom`.
+   */
+  @property({ reflect: true, attribute: 'position' }) position: Placement = 'bottom';
+
+  /**
+   * how listbox will display multiple items in toggle area:
+   * 'badge' for a badge with item count,
+   * 'chips' for a group of chips,
+   * '' for # items selected text (default)
+   */
+  @property({ attribute: 'selected-items-display', type: String }) selectedItemsDisplay: PfSelectItemsDisplay = '';
+
+  /**
+   * whether listbox controlled by combobox that supports typing
+   */
+  @property({ attribute: 'typeahead', type: Boolean }) typeahead = false;
+
+  @query('pf-chip-group') private _chipGroup?: PfChipGroup;
+  @query('pf-select-list') private _listbox?: PfSelectList;
+  @query('#toggle-input') private _input?: HTMLInputElement;
+  @query('#toggle-button') private _toggle?: HTMLButtonElement;
+
+  #toggle?: ToggleController;
+
+  #createOption!: PfSelectOption;
+
+  /**
+   * label for toggle button
+   */
+  get #buttonLabel() {
+    return this.hasBadge || this.hasChips ?
+      this.defaultText : this.#isMulti ?
+      `${this.#valueTextArray.length} ${this.itemsSelectedText}` : this.#valueText.length > 0 ?
+        this.#valueText : this.defaultText;
+  }
+
+  /**
+   * whether listbox is aria-multiselectable
+   */
+  get #isMulti() {
+    return this.multiSelectable || this.hasCheckboxes;
+  }
+
+  /**
+   * listbox template
+   */
+  get #selectList() {
+    const checkboxes = this.hasCheckboxes ? 'checkboxes' : false;
+    const { height, width } = this.getBoundingClientRect() || {};
+    const styles = this.alwaysExpanded ? '' : `margin-top: ${height || 0}px;width: ${width || 'auto'}px`;
+    return html`
+      <pf-select-list 
+        id="listbox" 
+        style="${styles}"
+        class="${classMap({ checkboxes })}"
+        aria-disabled="${this.disabled ? 'true' : 'false'}"
+        ?hidden=${!this.alwaysExpanded && !this.expanded}
+        ?case-sensitive=${this.caseSensitive}
+        ?disable-filter="${this.disableFilter}"
+        ?match-anywhere=${this.matchAnywhere}
+        ?multi-selectable=${this.#isMulti}
+        @input=${this.#onListboxInput}
+        @change=${this.#onListboxChange}
+        @listboxoptions=${this.#updateValueText}
+        @select=${this.#onListboxSelect}
+        @optioncreated=${this.#onOptionCreated}>
+        <slot></slot>
+      </pf-select-list>`;
```

Comment:
```suggestion
      <pf-select-list id="listbox" 
          style="${styles}"
          class="${classMap({ checkboxes })}"
          .disabled="${this.disabled}"
          .hidden="${!this.alwaysExpanded && !this.expanded}"
          .caseSensitive="${this.caseSensitive}"
          .disableFilter="${this.disableFilter}"
          .matchAnywhere="${this.matchAnywhere}"
          .multiSelectable="${this.#isMulti}"
          @input="${this.#onListboxInput}"
          @change="${this.#onListboxChange}"
          @listbox-options=${this.#updateValueText}
          @select=${this.#onListboxSelect}
          @option-created=${this.#onOptionCreated}>
        <slot></slot>
      </pf-select-list>`;
```

Reply:
do we need select-list?

if it's only for pf-select, why not use directives or just html and js?
if it's for other elements as well, why not `<pf-listbox>`?

we don't need to 1:1 match react's dev-facing apis